src/Ledger/Ledger/Properties.agda

```diff
@@ -82,29 +87,536 @@ instance
+
+    utxo-govst-connex : {props : List GovProposal}
+      → dom (proposalDepositsΔ props pp txb ˢ) ≡ᵉ fromList (map f (updateGovStates [] (map inj₂ props)))
+    utxo-govst-connex {[]} = dom∅
+
+    utxo-govst-connex {p ∷ ps} = let
+      open SetoidReasoning (≡ᵉ-Setoid{DepositPurpose})
+      n = length ps
+      n' = length{A = GovVote ⊎ GovProposal} (map inj₂ ps) in
+      begin
+      dom (proposalDepositsΔ (p ∷ ps) pp txb ˢ)
+        ≈⟨ dom∪⁺ ⟩
+      dom (updateProposalDeposits ps txid govActionDeposit ∅) ∪ dom (❴ GovActionDeposit (txid , length ps) , govActionDeposit ❵ ˢ)
+        ≈⟨ ∪-cong (utxo-govst-connex{ps}) dom-single≡single ⟩
+      fromList (map f (updateGovStates [] (map inj₂ ps))) ∪ fromList [ GovActionDeposit (txid , length ps) ]
+        ≈⟨ ∪++ ⟩
+      fromList (map f (updateGovStates [] (map inj₂ ps)) ++ map f [ (mkAction p (length ps)) ])
+        ≈⟨ ≡ᵉ.reflexive (cong fromList (sym (map-++ f (updateGovStates [] (map inj₂ ps)) [ (mkAction p (length ps)) ]))) ⟩
+      fromList (map f ((updateGovStates [] (map inj₂ ps)) ∷ʳ (mkAction p (length ps))))
+        ≈⟨ ≡ᵉ.reflexive (cong fromList (map-++ f (updateGovStates [] (map inj₂ ps)) [ mkAction p n ])) ⟩
+      fromList ((map f (updateGovStates [] (map inj₂ ps)) ++ map f [ mkAction p n ]))
+        ≈˘⟨ ∪++ ⟩
+      fromList (map f (updateGovStates [] (map inj₂ ps))) ∪ fromList (map f [ mkAction p n ])
+        ≈˘⟨ ∪-comm (fromList (map f [ mkAction p n ])) (fromList (map f (updateGovStates [] (map inj₂ ps)))) ⟩
+      fromList (map f [ mkAction p n ]) ∪ fromList (map f (updateGovStates [] (map inj₂ ps)))
+        ≈⟨ ∪++ ⟩
+      fromList (map f [ mkAction p n ] ++ map f (updateGovStates [] (map inj₂ ps)))
+        ≈⟨ ≡ᵉ.reflexive (cong fromList (map-++ f [ mkAction p n ] (updateGovStates [] (map inj₂ ps)))) ⟩
+      fromList (map f ([ mkAction p n ] ++ updateGovStates [] (map inj₂ ps)))
+        ≈˘⟨ ≡ᵉ.reflexive (cong fromList (cong (λ x → map f ([ mkAction p x ] ++ updateGovStates [] (map inj₂ ps))) (length-map inj₂ ps))) ⟩
+      fromList (map f ([ mkAction p n' ] ++ updateGovStates [] (map inj₂ ps)))
+        ≈˘⟨ ≡ᵉ.reflexive (cong fromList (updateGovStates≡ (map inj₂ ps))) ⟩
+      fromList (map f (updateGovStates [] (inj₂ p ∷ map inj₂ ps)))
+        ∎
+
+
+  module _
+    (tx : Tx) (let open Tx tx renaming (body to txb)) (let open TxBody txb)
+    (Γ : LEnv) (let open LEnv Γ renaming (pparams to pp))
+    (s : LState)
+    where
+
+    -- MAIN THEOREM --
+    LEDGER-govDepsMatch : ∀ {s' : LState} → Γ ⊢ s ⇀⦇ tx ,LEDGER⦈ s'
+                          → govDepsMatch s → govDepsMatch s'
+
+    LEDGER-govDepsMatch (LEDGER-I⋯ tx-not-valid
+                          (UTXOW-inductive⋯ _ _ _ _ _ _ _
+                            (UTXO-inductive⋯ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ UTXOS-sts)))
+      aprioriMatch with UTXOS-sts
+    ... | (_⊢_⇀⦇_,UTXOS⦈_.Scripts-Yes u) = contradiction (trans (sym $ proj₂ u) tx-not-valid) (λ ())
+    ... | (_⊢_⇀⦇_,UTXOS⦈_.Scripts-No _) = aprioriMatch
```

Comment:
Like before, assuming you have the pattern defined, you can do this:
```suggestion
    LEDGER-govDepsMatch (LEDGER-I⋯ refl (UTXOW-UTXOS (Scripts-No _))) aprioriMatch = aprioriMatch
```

Reply:
Very nice 👍🏼